src/ergw_aaa_radius.erl

```diff
@@ -737,8 +756,34 @@ radius_accounting_opts() ->
 		       ?Acct_Status_Type,
 		       ?Event_Timestamp]).
 
-send_request(Req, Session, #{server := NAS} = Opts) ->
+send_request(Req, Session, #{server := NAS} = Opts)
+  when is_tuple(NAS)->
     Id = maps:get('NAS-Identifier', Session, <<"NAS">>),
     RadiusClientOpts = [{client_name, maps:get(client_name, Opts, Id)},
-			{server_name, maps:get(server_name, Opts, Id)}],
-    eradius_client:send_request(NAS, Req, RadiusClientOpts).
+			{server_name, maps:get(server_name, Opts, Id)},
+			{retries, maps:get(retries, Opts, 3)},
+			{timeout, maps:get(timeout, Opts, 5000)}],
```

Comment:
if you preload the config defaults with those options, using a simple maps match in the function header is sufficient.

Reply:
config  defaults  added and maps pattern match where appropriate